### PR TITLE
Chunk multicall

### PR DESCRIPTION
**Pull request by yabirgb**
We are using multicall contract in the ENS queries to obtains ENS names for addresses. The problem that
we observed is that when the number of addresses is high (I tested it with 73) the url of the call to
the nodes was so long that it failed to make the request. The solution I propose chunks the query arguments
to the multicall contract to avoid such urls. I've tried different chunk sizes and 20 was the biggest that made
the query work.

Also I've avoided to repeat the same logic for the `multicall_2` contract since we don't use it in a way that
the logic breaks in the same way and could affect the speed of some queries.

### CODE REVIEW

File: rotkehlchen/chain/ethereum/modules/uniswap/utils.py [hunk 1]
```diff
@@ -191,7 +191,7 @@ def find_uniswap_v2_lp_price(
     decoded = []
     for (method_output, method_name) in zip(output, methods):
         call_success = True
-        if multicall_method == multicall_2:
+        if multicall_method == multicall_2:  # pylint: disable=comparison-with-callable
```
Comment (nebolax) [inline rotkehlchen/chain/ethereum/modules/uniswap/utils.py:194]: Are you sure that you need it? Everything works fine without this, doesn't it?

**Review by nebolax — COMMENTED** [review]
The problem that you are describing seems to be only about ENS names, not all `multicall`s. So maybe do one of:
1. Split calls into chunks only where ENS names are called?
2. Make chunking optional (so `multicall` function will be chunking only if a specific argument is passed)?

**Comment by nebolax** [comment]
Alright!
> @nebolax Talking with lefteris we said that all the multicalls should take care of this since is something that can happen if the author doesn't know about this limitation

**Review by LefterisJP — COMMENTED** [review]
Have you tested it? Do you think it would also make sense to do an actual query length check? It depends on the type of the parameters not only on the size of the chunks.

It the call has only 1 int parameter, then it can have more chunks. If each call has multiple addresses then less, if the parameters are dynamic ... welll we are screwed.

We already do this logic in chain/ethereum/tokens.py btw. Do you think it would make sense to combine this?

File: rotkehlchen/chain/ethereum/modules/uniswap/utils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (yabirgb) [reply rotkehlchen/chain/ethereum/modules/uniswap/utils.py]: The reason is that the arguments differ and then the linter warned us but with an error I wasn't expecting and didn't relate

**Comment by yabirgb** [comment]
@LefterisJP Yes, I took my approach from there. I tried to solve this problem limiting the length of the URL to 2000 chars (the max allowed in the standard) but found that chunking and the combining the request after in a deeper level was very complicated since you don't know exactly the format of the response of the contract and how to group it. I ended up with a very complicated logic for managing the outputs and it was not working properly. 

This is not the best approach but at least it makes us to be sure it will work without breaking anything else. I've tested it with the big account and queried the information without any issue trying different nodes. Also tried other parts of the app where the multicall is used and had no issue.

We can create a follow up of this issue to improve the code and do the check for chunking when calling the node as I was doing before, but as I said, for how the response of the contracts are I ended up with a complicated logic. Maybe with more time something easier comes u[p

File: rotkehlchen/chain/ethereum/utils.py [hunk 2]
```diff
@@ -130,6 +137,8 @@ def multicall_2(
         require_success: bool,
         call_order: Optional[Sequence['NodeName']] = None,
         block_identifier: BlockIdentifier = 'latest',
+        # only here to comply with multicall
+        calls_chunk_size: int = MULTICALL_CHUNKS,  # pylint: disable=unused-argument
```
Comment (LefterisJP) [inline rotkehlchen/chain/ethereum/utils.py:141]: why not implement chunking in that one too?

**Review by LefterisJP — APPROVED** [review]
I am going to provisionally approve.

Not super happy with this. Lots of code smells.

For example adding the un-needed argument is imo unnecessary. Better have type: ignore in the check if there is no need for the types to be the same ..

**Pull request merged**